File: mas-frontend/frontend/src/types/agent.ts
export interface ToolCall {
  tool_name: string;
  tool_type: string;
  parameters: Record<string, any>;
  result: Record<string, any>;
  duration_ms: number;
  timestamp: Date;
  success: boolean;
}

export interface AgentResponse {
  agent_name: string;
  agent_type: string;
  response: string;
  tools_used: ToolCall[];
  processing_time_ms: number;
  timestamp: Date;
}

export interface ExecutionTrace {
  session_id: string;
  request_id: string;
  user_message: string;
  coordinator_response: string;
  agent_sequence: string[];
  total_time_ms: number;
  agent_responses: AgentResponse[];
  timestamp: Date;
}

export interface AgentInfo {
  name: string;
  type: string;
  description: string;
  tools: string[];
  color: string;
  icon: string;
}

File: mas-frontend/backend/app/models/agent.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime

class ToolCall(BaseModel):
    tool_name: str
    tool_type: str
    parameters: Dict[str, Any]
    result: Dict[str, Any]
    duration_ms: float
    timestamp: datetime
    success: bool = True
    
    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat()
        }

class AgentResponse(BaseModel):
    agent_name: str
    agent_type: str
    response: str
    tools_used: List[ToolCall]
    processing_time_ms: float
    timestamp: datetime
    
    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat()
        }

class AgentInfo(BaseModel):
    name: str
    type: str
    description: str
    tools: List[str]
    color: str
    icon: str
    model: Optional[str] = None